hyperspy/io.py

```diff
@@ -47,6 +47,42 @@
     "\t\tPath: %s")
 
 
+def _infer_file_reader(extension):
+    """Return a file reader from the plugins list based on the file extension.
+
+    If the extension is not found or understood, returns
+    the Python imaging library as the file reader.
+
+    Parameters
+    ----------
+    extension : str
+        File extension, without initial "." separator
+
+    Returns
+    -------
+    reader : func
+        The inferred file reader.
+
+    """
+    rdrs = [rdr for rdr in io_plugins if extension.lower() in rdr.file_extensions]
+
+    if not rdrs:
+        # Try to load it with the python imaging library
+        _logger.warning(
+            f"Unable to infer file type from extension '{extension}'. "
```

Comment:
This warning will also be used when providing `reader='emd'` for example, which may not be the most appropriate but we can leave it as it is for now and improve it in a separate PR.

Reply:
Ok!